javascript/javascript1/week3/script.js

```diff
@@ -0,0 +1,191 @@
+console.log("=== Assignment 1 ===")
+
+const names = [
+    "Peter",
+    "Ahmad",
+    "Yana",
+    "kristina",
+    "Rasmus",
+    "Samuel",
+    "katrine",
+    "Tala",
+  ];
+const nameToRemove = "Ahmad";
+  
+// RemoveName
+let RemoveName=(nameToRemove, names)=>{
+    for(let i=0; i<names.length; i++){
+        if (names[i]===nameToRemove){
+            // splice
+            names.splice(i,1)
+            break
+           
+        }
+      }  
+ }
+
+RemoveName(nameToRemove,names)
+console.log(names); 
+
+
+
+console.log("=== Assignment 2 ===")
+
+// TravelTime
+let TravelTime=(t_Info)=>{
+    let timeInHours = (t_Info.destinationDistance / t_Info.speed);
+    let hours = Math.floor(timeInHours);
+    let minutes = Math.round((timeInHours - hours) * 60)
+    return `${hours} hours and ${minutes} minutes`
+}
+const travelInformation = {
+    speed: 50,
+    destinationDistance: 432,
+};
+const travelTime = TravelTime(travelInformation);
+console.log(travelTime);
+
+
+
+console.log("=== Assignment 3 ===")
+const seriesDurations = [
+    {
+      title: "Game of thrones",
+      days: 3,
+      hours: 1,
+      minutes: 0,
+    },
+    {
+      title: "Sopranos",
+      days: 3,
+      hours: 14,
+      minutes: 0,
+    },
+    {
+      title: "The Wire",
+      days: 2,
+      hours: 12,
+      minutes: 0,
+    },
+  ];
+
+let logOutSeriesText=(seriesDurations)=>{
+    const age=80
+    const year=365
+    const hours=24
+    const minutes=60
+
+    let total_life_span_in_minutes=age * year * hours * minutes
+    let total_percentage=0
+
+    seriesDurations.forEach(series => {
+        const series_minutes = (series.days * 24 * 60) + (series.hours * 60) + series.minutes;
+        const percentage = (series_minutes / total_life_span_in_minutes) * 100;
+        total_percentage += percentage;
+        console.log(`${series.title} took ${percentage.toFixed(3)}% of my life`);
+      });
+    
+      console.log(`\nIn total that is ${total_percentage.toFixed(3)}% of my life`);
+    }
+  
+logOutSeriesText(seriesDurations)
+
+// for experiment
+// seriesDurations.forEach((e) => {
+//     let fomate_sting=e.days+" "+e.hours +" "+e.minutes
+//     console.log(fomate_sting)
+// })
+
+
+
+console.log("=== Assignment 4 ===")
+
+const notes = [];
+
+// saveNote
+let saveNote=(content, id)=>{
+  let note={
+    'content':content,
+    'id':id
+  }
+  notes.push(note)
```

Comment:
No need to define a variable, when you use it a single time.

When you have object keys that are the same names as the properties, you can define the entire thing a bit shorter:
```suggestion
  notes.push({ content, id})
```

Reply:
100% right but i am developing this practice to more clear about my object in terms of keys and values (can say i want to double sure about my object keys and values )

  let note={
    content,
    id
  } (done)